frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import * as _ from 'lodash';
+import { Form, FormControl, FormGroup, HelpBlock } from 'patternfly-react';
+
+import {Dropdown, ButtonBar, history, NameValueEditorPair} from './utils';
+import { makeReduxID } from './utils/firehose';
+import { k8sCreate } from './../module/k8s';
+import actions from './../module/k8s/k8s-actions';
+import { StorageClassModel } from './../models';
+import { NameValueEditor } from './utils/name-value-editor';
+import { Link } from 'react-router-dom';
+
+const defaultState = {
+  newStorageClass: {
+    name: '',
+    description: '',
+    type: null,
+    parameters: {},
+    reclaim: null
+  },
+  customParams: [['', '']],
+  validationSuccessful: false,
+  loading: false,
+  error: null,
+  fieldErrors: {parameters: {}}
+};
+
+/*eslint-disable no-undef*/
+export class StorageClassForm_ extends React.Component<StorageClassFormProps, StorageClassFormState> {
+
+  resources: Resources;
+  reduxId: string;
+  previousName: string;
+
+  constructor(props) {
+    super(props);
+    this.state = defaultState;
+    this.reduxId = makeReduxID(StorageClassModel, {});
+    this.props.watchK8sList(this.reduxId, {}, StorageClassModel);
+    this.resources;
+    this.previousName = '';
+  }
+
+  storageTypes = {
+    aws: {
+      title: 'AWS Elastic Block Storage',
+      provisioner: 'kubernetes.io/aws-ebs',
+      documentationLink: 'https://kubernetes.io/docs/concepts/storage/storage-classes/#aws-ebs',
+      parameters: {
+        type: {
+          name: 'Type',
+          values: {io1: 'io1', gp2: 'gp2', sc1: 'sc1', st1: 'st1'},
+          placeholder: 'Choose AWS Type'
+        },
+        zone: {
+          name: 'Zone',
+          placeholder: 'AWS zone',
+          validation: (params) => {
+            if (params.zone.value !== '' && _.get(params, 'zones.value', '') !== '') {
+              return 'Zone and zones parameters must not be used at the same time';
```

Comment:
For error messages and placeholder text that are complete sentence, add a period at the end.

Reply:
Done